eventsources/eventing.go

```diff
@@ -402,6 +467,23 @@ func (e *EventSourceAdaptor) run(ctx context.Context, servers map[apicommon.Even
 				}
 				if err = common.Connect(&backoff, func() error {
 					return s.StartListening(ctx, func(data []byte, opts ...eventsourcecommon.Options) error {
+						if filter, ok := filters[getFilterKey(s.GetEventSourceType(), s.GetEventName())]; ok {
+							dataMap := make(map[string]interface{})
+							err := json.Unmarshal(data, &dataMap)
+							if err != nil {
+								logger.Warn("unable to json unmarshal data")
+								return nil
+							}
+							proceed, err := shouldDispatch(filter.Expression, dataMap)
+							if err != nil {
+								logger.Warn("failed to publish event, filter condition not met")
```

Comment:
This is not a failure of dispatching. I think it might be better give a `Error` (Apologize if I have suggested `Warn` previously):

```
logger.Errorw("Failed to parse the filter", zap.Error(err))
```